### PR TITLE
feat: Add Distributed Tracing support for Azure Functions HTTPTrigger.

### PR DESCRIPTION
**Pull request by jaffinito**
## Description

Add support for accepting DT payloads for Azure Functions HTTPTrigger.  Does this only on the incoming request since DT is forward only.  Includes expanded HTTPTrigger integration tests that include DT headers and validation.

**Note:** Required a reference to Microsoft.Extensions.Primitives in order to get access to StringValues type.  I used the newest version of 2.x that was supported - many versions are deprecated - to reduce the chance of problems.

# Author Checklist
- [ ] Unit tests, Integration tests, and Unbounded tests completed
- [ ] Performance testing completed with satisfactory results (if required)

# Reviewer Checklist
- [ ] Perform code review
- [ ] Pull request was adequately tested (new/existing tests, performance tests)

### CODE REVIEW

File: src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AzureFunction/FunctionDetails.cs [hunk 1]
```diff
@@ -0,0 +1,196 @@
...
+        }
+
+        // don't parse request parameters here if the Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore assembly is loaded.
+        // If it is loaded, parsing occurs over in FunctionsHttpProxyingMiddlewareWrapper
+        if (_hasAspNetCoreExtensionsReference.Value)
+        {
+            return;
+        }
+
+        object features = functionContext.Features;
```
Comment (tippmar-nr) [inline src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AzureFunction/FunctionDetails.cs:129]: This block of code also needs to look for DT headers - it gets called when the customer is not using the `Microsoft.Azure.Functions.Worker.Extensions.Http.AspNetCore` package. I believe we have an integration test that exercises this section of code.

Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Extensions/Providers/Wrapper/AzureFunction/FunctionDetails.cs:129]: I will take a look.

File: src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Helpers/DictionaryHelpers.cs [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Collections.Generic;
+using Newtonsoft.Json;
+
+namespace NewRelic.Agent.Extensions.Helpers
+{
+    public static class DictionaryHelpers
+    {
+        public static IReadOnlyDictionary<string, object> FromJson(string json)
```
Comment (tippmar-nr) [inline src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Helpers/DictionaryHelpers.cs:11]: Codecov wants a unit test for this helper...

**Review by tippmar-nr — DISMISSED** [review]
Looks good!

File: src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Helpers/DictionaryHelpers.cs (hunk 2 above)
Comment (jaffinito) [reply src/Agent/NewRelic/Agent/Extensions/NewRelic.Agent.Extensions/Helpers/DictionaryHelpers.cs:11]: I am working on a test.

**Review by tippmar-nr — APPROVED** [review]

**Review by nrcventura — APPROVED** [review]

**Pull request merged**